crates/turbopack-node/src/evaluate.rs

```diff
@@ -29,17 +32,67 @@ use crate::{
     bootstrap::NodeJsBootstrapAsset,
     embed_js::embed_file_path,
     emit, emit_package_json,
-    pool::{NodeJsPool, NodeJsPoolVc},
-    EvalJavaScriptIncomingMessage, EvalJavaScriptOutgoingMessage, StructuredError,
+    pool::{NodeJsOperation, NodeJsPool, NodeJsPoolVc},
+    StructuredError,
 };
 
+#[derive(Serialize)]
+#[serde(tag = "type", rename_all = "camelCase")]
+enum EvalJavaScriptOutgoingMessage<'a> {
+    #[serde(rename_all = "camelCase")]
+    Evaluate { args: Vec<&'a JsonValue> },
+}
+
+#[derive(Deserialize)]
+#[serde(tag = "type", rename_all = "camelCase")]
+enum EvalJavaScriptIncomingMessage {
+    FileDependency {
+        path: String,
+    },
+    BuildDependency {
+        path: String,
+    },
+    DirDependency {
+        path: String,
+        glob: String,
+    },
+    EmittedError {
+        severity: IssueSeverity,
+        error: StructuredError,
+    },
+    Value {
+        // TODO: There is like 3 levels of JSON string encoding that goes into returning a value,
+        // making it really inefficient.
+        data: String,
+    },
+    End {
+        data: Option<String>,
+    },
+    Error(StructuredError),
+}
+
+#[derive(Debug)]
+enum LoopResult {
```

Comment:
This is similar to the [ControlFlow](https://doc.rust-lang.org/stable/std/ops/enum.ControlFlow.html) enum, maybe its name could reflect that? e.g. `OperationControlFlow`.